component/block_cache/block_cache.go

```diff
@@ -242,17 +243,30 @@ func (bc *BlockCache) Configure(_ bool) error {
 	bc.tmpPath = ""
 	if conf.TmpPath != "" {
 		bc.tmpPath = common.ExpandPath(conf.TmpPath)
+		//check mnt path is not same as temp path
+
+		err = config.UnmarshalKey("mount-path", &bc.mntPath)
+		if err == nil && bc.mntPath == bc.tmpPath {
```

Comment:
Test case for error scenarios are missing

Reply:
done